File: src-tauri/src/models/provider.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize)]
pub struct ProviderSummary {
    pub id: String,
    pub name: String,
    pub base_url_masked: String,
    pub api_key_masked: String,
    pub interface_type: String,
    pub status: String,
    pub model_count: i64,
    pub last_checked_at: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct CreateProviderInput {
    pub name: String,
    pub base_url: String,
    pub api_key: Option<String>,
    pub interface_type: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct UpdateProviderInput {
    pub name: String,
    pub base_url: String,
    pub api_key: Option<String>,
    pub interface_type: String,
}

#[derive(Debug, Clone, Serialize)]
pub struct ModelSummary {
    pub id: String,
    pub provider_id: String,
    pub name: String,
    pub model_type: String,
    pub capabilities: Vec<String>,
    pub supports_streaming: bool,
    pub recommended_concurrency: Option<i64>,
}

#[derive(Debug, Clone, Serialize)]
pub struct ProviderModelScanResult {
    pub provider_id: String,
    pub models: Vec<ModelSummary>,
    pub message: String,
    pub scanned_at: String,
}

#[derive(Debug, Clone, Serialize)]
pub struct DeleteResult {
    pub id: String,
    pub deleted: bool,
}

#[derive(Debug, Clone, Serialize)]
pub struct ProviderConnectionResult {
    pub provider_id: String,
    pub ok: bool,
    pub status: String,
    pub message: String,
    pub checked_at: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct ProviderDiagnosticsInput {
    pub provider_id: String,
    pub model_id: Option<String>,
    pub dataset_id: Option<String>,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct ProviderDiagnosticsResult {
    pub provider_id: String,
    pub status: String,
    pub checked_at: String,
    pub engine_mode: String,
    pub endpoints: Vec<DiagnosticEndpoint>,
    pub recommendations: Vec<String>,
}

#[derive(Debug, Clone, Deserialize, Serialize)]
pub struct DiagnosticEndpoint {
    pub name: String,
    pub method: String,
    pub path: String,
    pub ok: bool,
    pub latency_ms: Option<i64>,
    pub http_status: Option<i64>,
    pub message: String,
    pub error_kind: Option<String>,
}

#[derive(Debug, Clone)]
pub struct ProviderConnectionConfig {
    pub id: String,
    pub name: String,
    pub base_url: String,
    pub api_key_plaintext: String,
    pub interface_type: String,
}

#[derive(Debug, Clone)]
pub struct DiscoveredModel {
    pub name: String,
    pub model_type: String,
    pub capabilities: Vec<String>,
    pub supports_streaming: bool,
    pub recommended_concurrency: Option<i64>,
}

File: src/types/api/provider.ts
export const PROVIDER_INTERFACE_TYPES = [
  "OpenAI",
  "OpenAI-Response",
  "Anthropic",
  "Gemini",
  "Jina Rerank",
] as const;

export type ProviderInterfaceType = (typeof PROVIDER_INTERFACE_TYPES)[number];

export type ProviderSummary = {
  id: string;
  name: string;
  base_url_masked: string;
  api_key_masked: string;
  interface_type: string;
  status: string;
  model_count: number;
  last_checked_at: string | null;
  created_at: string;
};

export type CreateProviderInput = {
  name: string;
  base_url: string;
  api_key?: string;
  interface_type: ProviderInterfaceType;
};

export type UpdateProviderInput = {
  name: string;
  base_url: string;
  api_key?: string;
  interface_type: ProviderInterfaceType;
};

export type ProviderImportItem = {
  name: string;
  base_url: string;
  api_key?: string;
  interface_type: ProviderInterfaceType;
  models?: string[];
};

export type ProviderImportInput = {
  items: ProviderImportItem[];
};

export type ProviderImportItemResult = {
  index: number;
  status: "created" | "skipped" | "failed" | string;
  provider_id: string | null;
  message: string;
};

export type ProviderImportResult = {
  created: number;
  skipped: number;
  failed: number;
  items: ProviderImportItemResult[];
};

export type DeleteResult = {
  id: string;
  deleted: boolean;
};

export type ProviderConnectionResult = {
  provider_id: string;
  ok: boolean;
  status: string;
  message: string;
  checked_at: string;
};

export type ProviderDiagnosticsInput = {
  provider_id: string;
  model_id?: string;
  dataset_id?: string;
};

export type DiagnosticEndpoint = {
  name: string;
  method: string;
  path: string;
  ok: boolean;
  latency_ms: number | null;
  http_status: number | null;
  message: string;
  error_kind: string | null;
};

export type ProviderDiagnosticsResult = {
  provider_id: string;
  status: "passed" | "warning" | "failed" | "unsupported" | string;
  checked_at: string;
  engine_mode: string;
  endpoints: DiagnosticEndpoint[];
  recommendations: string[];
};
